packages/gatsby/src/query-runner/query-compiler.js

```diff
@@ -50,7 +50,8 @@ class Runner {
   }
 
   async parseEverything() {
-    let files = await globp(`${this.baseDir}/**/*.js`)
+    let files = await globp(`${this.baseDir}/**/*.+(t|j)s?(x)`)
+    files = files.filter(d => !d.match(/(\.stories\.|\.test\.|\.d\.)/))
```

Comment:
this (I think) is too prescriptive. you'll undoubtably catch actual source files. perhaps it makes sense to make a a config option for "ignore paths" or something that is specifiable in the gatsby-config